pkg/utils/segutils.go

```diff
@@ -194,3 +194,21 @@ func WriteValidityFile(segBaseDir string) error {
 
 	return nil
 }
+
+func GetSegDirFromSegKey(segKey string) string {
```

Comment:
Can you reuse `getSegBaseDirFromFilename()` (making it public if needed), or does that function not handle this case?

Reply:
addressed in the latest commit.